core/src/main/java/org/apache/james/core/quota/QuotaCurrentValue.java

```diff
@@ -0,0 +1,124 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+
+import org.apache.james.core.Username;
+
+import com.google.common.base.MoreObjects;
+
+public class QuotaCurrentValue {
+
+    public static class QuotaKey {
```

Comment:
Those 2 QuotaKey classes that look similar but a bit different is looking wrong and confusing to me. 

Or we:
- find a way to unify that into one POJO (and separated, not a subclass of an other POJO) => then maybe using QuotaScope as identifier and finding a good structure would make sense then
- just sticking to builders here (need to add them), and have the QuotaKey for the DAO with current values only (next PR), if sticking to Username as identifier.

I start to think though now that refactoring QuotaScope to fit the identifier and scope needs might be the better way here

Reply:
After reading Benoit's comment, I feel it is better to group these field in a subclass as a key. Not sure why you are confused.

Reply:
He suggested to do one class QuotaKey, not two?

Reply:
I am not sure what he mean exactly but it gives me the idea :))

Reply:
I think you go the idea wrong. The fact that @quantranhong1999 put a thumb up on the comment above means he likely agrees to the comment as well

Please rework on this :)